### PR TITLE
805 - add link error handling

### PR DESCRIPTION
**Pull request by jaskwong**
- catch errors returned from backend and display error message notification

### CODE REVIEW

**Comment by jaskwong** [comment]
Right now, I just caught the error for link adding, should I add try/catch for the other requests, I wasn't sure about the scope of this issue on asana

**Comment by wynnset** [comment]
Yes, please add it in for everything. It would also be nice if we show a user an alert saying what went wrong (I believe the backend sends an error message back in the response that we can show). In terms of implementation: 
- We might want to create a helper class for this so that we do not repeat code where possible
- It would be nice if we do not use alerts, but something less intrusive like [Bootstrap toasts](https://bootstrap-vue.org/docs/components/toast)

**Review by wynnset — COMMENTED** [review]
Very elegant solution, @jaskwong good job! I think we could improve this by doing more than just showing the error if the request fails, but that's probably outside the scope here. The exception is areas where there is already something implemented - those should be transferred to `actions.js`. For example, where `updateNodeCoordinates` is being called, it is resetting the location of the node - this should be transferred to `actions.js` along with any other instances (if any) where we are handling the exception more than just showing the error.

File: templates/vue/src/utils/errorHelper.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+    export default class ErrorHelper {
+    /**
+   * Return true if response is an error
+   *
+   * @param {Object} response
+   *
+   * @return {Boolean}
+   */
+    static isError(response) {
+        return response.status !== 200
```
Comment (rlnsy) [inline templates/vue/src/utils/errorHelper.js:10]: @jaskwong Should we support other successful response codes in the 200 range?

Comment (jaskwong) [reply templates/vue/src/utils/errorHelper.js]: good call, I'll update it, but right now this function is never called so I might delete it later

**Comment by rlnsy** [comment]
@jaskwong From what I can see, video errors from the backend are not being caught with this setup.  @shirleywbi has been working on video URL validation in the frontend, but we should probably add a catch-all handle with the same structure as you have here. Idk if this is outside the scope though.

**Comment by rlnsy** [comment]
Also, let me know if you want any help with the error-handling!

**Comment by jaskwong** [comment]
@rlnsy I think that would probably need a new error structure, the error handling on this branch is primarily for errors that happen as a result of the REST api when we try to update the database, also thanks for the offer, I think most of the error structure here is finished, and there's just a couple bugs that need to be worked out

File: templates/vue/src/components/TapestryNode.vue [hunk 2]
```diff
@@ -266,60 +268,52 @@ export default {
...
+              originalCoordinates.x == coordinates.x &&
+              originalCoordinates.y == coordinates.y
+            ) {
+              continue
             }
+            this.updateNodeCoordinates({
+              id,
+              coordinates,
+              originalCoordinates,
+            }).then(() => {
```
Comment (jaskwong) [inline templates/vue/src/components/TapestryNode.vue:314]: @narendrasss I'm not sure if you've had the chance to test yet, if not, do you mind changing this to async/await syntax I should've caught that earlier

**Review by narendrasss — APPROVED** [review]
Thanks @jaskwong this looks good! There was a small bug where the svg takes a long time to update when you drag nodes around. I pushed a small fix to address this.

Otherwise I think it's good to go 😄 We should make a note to add the appropriate tests for this in the near future.

**Review by wynnset — APPROVED** [review]

File: templates/vue/src/utils/errorHelper.js (hunk 1 above)
Comment (wynnset) [reply templates/vue/src/utils/errorHelper.js]: I've removed this function since I agree with @rlnsy and also it wasn't being used.

**Pull request merged**